### PR TITLE
DX-1459 Manage multi orgs in Operator

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

DX-1459

Note to reviewer: Better design of the diagrams will be provided

---

### Preview Link

<br>

[Key Concept - Operator User](https://deploy-preview-4903--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/key-concepts/key-concepts/)

[Key concepts - Operator Context](https://deploy-preview-4903--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/key-concepts/operator-context)

[Managing Multiple Orgs](https://deploy-preview-4903--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/getting-started/tyk-operator-multiple-organisations)

[Managing API Ownership](http://deploy-preview-4903--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/getting-started/tyk-operator-api-ownership/)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Added new guides on managing API ownership and multiple organizations with Tyk Operator.
- Introduced key concepts and Operator Context documentation for Tyk Operator.
- Updated links and descriptions in various documentation files.
- Restructured the menu for better navigation of Tyk Operator documentation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-operator-api-ownership.md</strong><dd><code>New guide on managing API ownership with Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/getting-started/tyk-operator-api-ownership.md

<li>Added a new guide on managing API ownership with Tyk Operator.<br> <li> Included YAML examples for defining <code>OperatorContext</code> and using <br><code>contextRef</code> in API definitions.<br> <li> Provided detailed explanations and practical examples.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-e9f0614eafcdbc0a472b177815d14b96c041c72cd54f6d07a67c00364f152249">+92/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-operator-multiple-organisations.md</strong><dd><code>New guide on managing multiple organizations with Tyk Operator</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/getting-started/tyk-operator-multiple-organisations.md

<li>Added a new guide on managing multiple organizations with Tyk <br>Operator.<br> <li> Included YAML examples for defining <code>OperatorContext</code> and using <br><code>contextRef</code> in API definitions.<br> <li> Provided best practices for multi-tenant API management.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-b314d2465630b34897bee4197937d7d5eba09e12bbc06824aca2d7e368435b01">+72/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>key-concepts.md</strong><dd><code>Added key concepts documentation for Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/key-concepts/key-concepts.md

<li>Added a new document explaining key concepts for Tyk Operator.<br> <li> Included sections on Operator User and multi-tenancy in Tyk Operator.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-afe9e0e893e3d8e8f172edd31e275626c29e3ca640ba9ff7ddc2e0c5735a89eb">+33/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>operator-context.md</strong><dd><code>Added documentation for Operator Context in Tyk Operator</code>&nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/key-concepts/operator-context.md

<li>Added a new document explaining the concept of <code>OperatorContext</code>.<br> <li> Included example scenarios and detailed explanations.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-83c877c9117372b19be80b5be1d37d40790865a977900bef3fa8e566f58b1a6f">+51/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>kubernetes.md</strong><dd><code>Updated link to Tyk Operator getting started guide</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-on-premises/kubernetes.md

- Updated the link to the Tyk Operator getting started guide.

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-611b9f8d99770cc880cc82ab76b35408b351a04124234a3870627ba0e35f5d49">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tyk-operator.md</strong><dd><code>Added alias and updated description for Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-operator.md

<li>Added an alias for the removed getting started overview page.<br> <li> Updated the description and image for Tyk Operator.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-98bfbaf10fee8c120d9fc042653aceef6e41137002f8cf4597d91a39506f2f32">+4/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ce-kubernetes.md</strong><dd><code>Updated link to Tyk Operator getting started guide</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-oss/ce-kubernetes.md

- Updated the link to the Tyk Operator getting started guide.

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-23f3c6cdc2bfec7a464086c7c0fe628a84b7828fa03b91150f8fc91628d702ac">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>getting-started-tyk-operator.md</strong><dd><code>Removed old getting started guide for Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-operator/getting-started-tyk-operator.md

- Removed the old getting started guide for Tyk Operator.

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-beda7f10a9337424b008bbca892fe0bd7ebda5eca4a10e970c06c4ccd5466c41">+0/-19</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Restructured menu for Tyk Operator documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

<li>Restructured the menu for Tyk Operator documentation.<br> <li> Added new entries for key concepts and getting started guides.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4903/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+24/-7</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+- API Keys
+- Users
+- Developers
+- Domains
+
+This structure is ideal for businesses with a complex hierarchy, where distinct departments operate independently but within the same overall infrastructure.
+
+### Tyk Operator
+
+Tyk Operator is a Kubernetes Controller that manages Tyk Custom Resources (CRs) such as API Definitions and Security Policies. Developers define these resources as CRDs, and Tyk Operator ensures that the desired state is reconciled with the Tyk Dashboard. This involves creating, updating, or deleting API configurations until the target state matches the desired state.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md]: Add Custom Resource Definitions (CRDs) into the second sentence and link to K8s - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/

File: tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    insecureSkipVerify: true
+    ingress:
+      httpPort: 8000
+      httpsPort: 8443
+    user_owners:
+    - a1b2c3d4e5f6
+    user_group_owners:
+    - 1a2b3c4d5e6f
+```
+
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md]: Add a summary explanation of what the key spec fields are, e.g mode, auth, org, url, user_owners and user_group_owners

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks @caroltyk PR LGTM! 

While in draft status, I have made minor structure changes to factor out key concepts into separate page and moved the guide into getting started menu folder.

File: tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md (hunk 2 above)
Comment (carolwkc) [reply tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md]: modified spec with comments

File: tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md (hunk 1 above)
Comment (carolwkc) [reply tyk-docs/content/tyk-stack/tyk-operator/tyk-operator-multiple-organisations.md]: updated

File: tyk-docs/content/product-stack/tyk-operator/key-concepts/key-concepts.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+This section provides an overview of the essential terminology and concepts used throughout the Tyk Operator documentation.
+
+## Operator User
+
+Tyk Operator is a Kubernetes Controller that manages Tyk Custom Resources (CRs) such as API Definitions and Security Policies. Developers define these resources as [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/), and Tyk Operator ensures that the desired state is reconciled with the Tyk Gateway or Dashboard. This involves creating, updating, or deleting API configurations until the target state matches the desired state.
+
+For the Tyk Dashboard, Tyk Operator functions as a system user, bound by Organisation and RBAC rules.
+
+During start up, Tyk Operator looks for these keys from `tyk-operator-conf` secret or from the environment variables and use these credentails to connect to Tyk.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-operator/key-concepts/key-concepts.md]:
```suggestion
During start up, Tyk Operator looks for these keys from `tyk-operator-conf` secret or from the environment variables (listed in the table below) and uses these credentials to connect to Tyk.
```

File: tyk-docs/data/menu.yaml [hunk 4]
```diff
@@ -3752,14 +3752,27 @@ menu:
         path: /tyk-operator
         category: Page
         show: True
-      - title: "Getting Started"
+      - title: "Key Concepts"
+        path: /product-stack/tyk-operator/key-concepts/key-concepts
         category: Directory
         show: True
         menu:
-        - title: "Overview"
```
Comment (dcs3spp) [inline tyk-docs/data/menu.yaml:3779]: Should the overview page be removed?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
@caroltyk PR LGTM!!!!! Not sure if still in draft have added minor suggestion and asked question if overview page should be removed

File: tyk-docs/data/menu.yaml (hunk 4 above)
Comment (carolwkc) [reply tyk-docs/data/menu.yaml:3779]: Yes it is removed

Comment (dcs3spp) [reply tyk-docs/data/menu.yaml:3779]: Thanks @caroltyk I've fixed the build errors

**Comment by sedkis** [comment]
Brilliant Carol.  This is excellent.  Thank you.

One thing we can think about adding:
1. What is the behaviour of the Tyk Operator do if NO Operator Context is provided in a situation where there are multiple orgs in the Tyk Dashboard.
2. What is the behaviour of the Tyk Operator if no ContextRef is provided in a CRD, yet multiple OperatorContexts are defined?

**Review by dcs3spp — APPROVED** [review]
Thanks @caroltyk PR and structure LGTM. Once peer approved and the diagrams have been improved as mentioned in review description, then ready for release.

**Comment by carolwkc** [comment]
> Brilliant Carol. This is excellent. Thank you.
> 
> One thing we can think about adding:
> 
> 1. What is the behaviour of the Tyk Operator do if NO Operator Context is provided in a situation where there are multiple orgs in the Tyk Dashboard.
> 2. What is the behaviour of the Tyk Operator if no ContextRef is provided in a CRD, yet multiple OperatorContexts are defined?

Added example scenarios [here](https://deploy-preview-4903--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/key-concepts/operator-context/#example-scenarios) to explain this.

**Comment by sedkis** [comment]
Amazing - Thank you. LGTM

**Comment by dcs3spp** [comment]
Marked PR as draft until diagrams are added

**Comment by carolwkc** [comment]
@dcs3spp diagrams updated and updated spelling of "organisation" to "organization"

**Pull request merged**